src/libs/sys-util-rs/src/cpu.rs

```diff
@@ -4,6 +4,55 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use std::collections::HashSet;
+
+use crate::sl;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid CPU list {0}")]
+    InvalidCpuList(String),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Parse cpuset in form of "1,3,5" and "1-4,6,8".
+pub fn split_cpus(cpus_string: &str) -> Result<Vec<u32>> {
+    let mut cpu_list = HashSet::new();
+
+    if !cpus_string.is_empty() {
+        let cpus_string_split: Vec<&str> = cpus_string.split(',').collect();
+        'next_token: for split_cpu in cpus_string_split {
+            let cpus: Vec<&str> = split_cpu.split('-').collect();
+            if cpus.len() == 1 {
+                let value = cpus[0];
+                if !value.is_empty() {
+                    if let Ok(cpu_id) = value.parse::<u32>() {
+                        cpu_list.insert(cpu_id);
+                        continue 'next_token;
+                    }
+                }
+            } else if cpus.len() == 2 {
+                if let Ok(start) = cpus[0].parse::<u32>() {
+                    if let Ok(end) = cpus[1].parse::<u32>() {
+                        for cpu in start..end + 1 {
```

Comment:
`start..=end` probably makes the intention here clearer.

Reply:
fixed